src/textEditor.ts

```diff
@@ -160,6 +160,14 @@ export class TextEditor {
     return word;
   }
 
+  static getTabCharacter(): string {
+    const { tabstop, expandtab } = configuration;
```

Comment:
It seems `tabstop` does not get updated on a per-document basis. So if VSCode auto-detects a file is tab-indented, or indented with 2 spaces instead of 4, this won't account for that. I wonder if there's a way to get that info.

Reply:
Maybe `TextEditorOptions.tabSize`? ([see here](https://code.visualstudio.com/api/references/vscode-api#TextEditorOptions))

Reply:
I changed it to use `TextEditorOptions` which seems to work. We should probably change other instances, but this is fine for now. Could you take a look and make sure this is reasonable? Thanks!

Reply:
should the information in the `configuration` object be stored per-document as well?

Reply:
ah I see. configuration is instantiated only once.

I'm not sure about making `getTabCharacter()` requires a `VimState` object, perhaps we can make the `TextEditor` class aware of the vim state by itself?

Reply:
I'm not sure I love that, since `TextEditor` is more of a module than a class (it probably *should* be a module, actually) and there are multiple `VimState`s floating around. Maybe it's better to just pass the editor itself into the function, since that's all we need?

Reply:
Then that makes sense to me